docs/key-issues/multiple_architectures.md

```diff
@@ -0,0 +1,266 @@
+# Platforms with multiple CPU architectures
+
+One important subset of ABI concerns is the CPU architecture for which a binary
+artefact has been built. Attempting to run a binary on hardware that doesn't
+match the CPU architecture (or architecture variant[^1]) for which the binary
+was built will generally lead to crashes, even if the ABI being used is
+otherwise compatible.
+
```

Comment:
I think we're back to the key problem being support for iOS and Android (the one other concrete case of aarch64 variants like SBSA may not be an issue after all); so tweaking the intro here would be good I think to state that clearly. We don't have concrete problems for any other platforms, because single-arch wheels are always preferred.

Reply:
I disagree. macOS is also firmly in this grouping, at least for my use case. 

This isn't something that affects the "Conda/System python" use case, because you're in control of the Python interpreter, and as the person installing a binary package, you're pulling binaries that are specific to your usage requirement, which is a single CPU architecture by definition.

However, if you're building a redistributable macOS app, you *must* care about all the architectures that your user base may be using. If I build an app on M1, and give it to a user on x86, I still want my app to run. The set of problems here are identical to those on mobile platforms. The Python ecosystem has *a* solution - universal wheels - but this solution is (a) not inherently useful for mobile platforms without significant changes, and (b) not liked by some segments of the desktop user base.

From a packaging perspective, I distinctly *do not* prefer single arch wheels, as they complicate the process of building an app - at least, it does given the current state of Python packaging tooling. If the Python ecosystem is going to converge on single architecture wheels as the only/preferred approach, then I'd argue that needs to be combined with a clear understanding of how use cases on platforms that need to support multiple architectures interact with that ecosystem. Being able to do "pip install" specifying multiple architectures and implicitly do a "delocate-fuse" (or equivalent) for each target architecture seems like a bare minimum.

Reply:
I guess I'm unclear why this is a `py2app` (or, in my case, `briefcase`) issue (beyond the obvious "it's your problem, so fix it" answer). As discussed further up the page, this isn't something that can be trivially resolved by end-user tooling. It's something that requires policies and annotation at the level of the package ecosystem - at the very least to describe how package merging *should* occur, if not to provide at least a reference implementation of that merging process.

Reply:
Agreed on the improvements needed discussed in the other thread.

> I guess I'm unclear why this is a `py2app` (or, in my case, `briefcase`) issue (beyond the obvious "it's your problem, so fix it" answer).

Beyond the "those who need it tend to have to do the work", it is because from a holistic design perspective it is the correct place to implement this. The only users that need this are `py2app`, `briefcase` & co, so let's support the use case there rather than both every single package author with more work. It also will scale better in practice that way (only 50% of package authors uploading fat wheels won't be enough for your needs here).

Reply:
>  It should be shared first-class tooling in the Python ecosystem that can be utilised by anyone that needs it.

I agree, and I like this. Let's state it like this in the "Potential solutions" section?

Ideally what should happen here is that this ends up in `auditwheel` and that becomes the one-stop shop for wheel distribution/packager needs, rather than having separate per-OS projects with patchy support for this kind of thing.

Reply:
I've added an entry under Potential solutions for this. Please resolve if that looks good.